storage/storage_dest.go

```diff
@@ -457,7 +487,7 @@ func (s *storageImageDestination) computeID(m manifest.Manifest) string {
 				continue
 			}
 			blobSum := m.FSLayers[i].BlobSum
-			diffID, ok := s.uncompressedOrTocDigest[blobSum]
+			diffID, ok := s.blobDiffIDs[blobSum]
```

Comment:
what error exactly do you have in mind?

If I try to pull a zstd image with `--format v2s1`, the operation fails with:

```
FATA[0005] creating an updated image manifest: Error during manifest conversion: "application/vnd.oci.image.layer.v1.tar+zstd": zstd compression is not supported for docker images 
```

Reply:
That’s a good point.

Still… what happens when someone creates an image with a gzip MIME type but a TOC annotation?

Reply:
do you suggest returning an explicit error if `!ok` (instead of returning an empty string) and propagating it to the caller?

Reply:
Actually the fallback code (randomly generating a new image ID and not reusing an image) is not too bad … just a comment to the effect that “this can, in principle, legitimately happen when a layer is reused by TOC” would be useful.